src/Data/Poly/Interpolation.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE CPP #-}
+
+module Data.Poly.Interpolation
+  ( lagrange
+  , hermite
+  ) where
+
+#if __GLASGOW_HASKELL__ < 910
+import Data.Foldable (foldl')
+#endif
+
+import Data.Poly.Internal.Dense
+import qualified Data.Vector.Generic as G
+
+-- | Compute the [Lagrange interpolating polynomial](https://en.wikipedia.org/wiki/Lagrange_polynomial).
+--
+-- This is the (unique) polynomial of minimal degree interpolating the given points.
+-- The values are given as @(x, y)@ pairs where @y@ is the value at @x@. The @x@ values must be distinct.
```

Comment:
Could we pass it as `Map a a` to guarantee that values are distinct? Requiring `Ord a` should not be too constraining...